build/misc.c

```diff
@@ -12,72 +15,36 @@
 #define BUF_CHUNK 1024
 
 struct StringBufRec {
-    char *buf;
-    char *tail;     /* Points to first "free" char */
-    int allocated;
-    int free;
+    std::string buf;
 };
 
 StringBuf newStringBuf(void)
 {
-    StringBuf sb = (StringBuf)xmalloc(sizeof(*sb));
-
-    sb->free = sb->allocated = BUF_CHUNK;
-    sb->buf = (char *)xcalloc(sb->allocated, sizeof(*sb->buf));
-    sb->buf[0] = '\0';
-    sb->tail = sb->buf;
-    
-    return sb;
+    return new StringBufRec {};
 }
 
 StringBuf freeStringBuf(StringBuf sb)
 {
-    if (sb) {
-	sb->buf = _free(sb->buf);
-	sb = _free(sb);
-    }
-    return sb;
+    delete sb;
```

Comment:
I assume / hope there's an intention to move away from `new` / `delete` at a later stage of refactoring?

Reply:
Oh, absolutely. This is really just stage 1 of the C++ onboarding, with many more to follow as time permits. The early steps look something like this:

Stage 1: natively allocate everything to allow use of non-trivial structures within
Stage 2: make our structs behave in STL containers (ie add destructors and copy-control)
Stage 3: take advantage of smart pointers and similar (look to replace manual reference counting, mutex locking and such)
...

And of course, look for opportunities to move stuff over to STL types is an on-going thing. I've been eliminating explicit new/delete where easily achieved when moving to STL containers. Stage 1 is sticking with these manual allocations just to keep things simple, just moving to STL containers presents plenty enough of unexpected side-effects :smile:

Reply:
There will also need to be stuff like mass-moving to the C++-versions of C headers, converting NULL to nullptr and whatnot.

And I really need to write this stuff down someplace where people can see...